core/src/View/Kokkos_ViewTraits.hpp

```diff
@@ -586,6 +583,22 @@ struct RemoveAlignedMemoryTrait {
  public:
   using type = typename TypeListToViewTraits<D, new_type_list>::type;
 };
+
+// Customization point for view hooks; default is to use the explicit template
+// parameter, but this can be customized to get the view hook from a special
+// memory space for example
+template <class DataType, class... Properties>
+constexpr auto customize_view_hooks() {
+  using traits_type = ViewTraits<DataType, Properties...>;
+  if constexpr (!std::is_void_v<typename traits_type::hooks_policy>) {
+    return typename traits_type::hooks_policy{};
+  }
+}
+
+template <class DataType, class... Properties>
+struct ViewHooksFromTraits {
```

Comment:
Why is this struct needed? Could the view hook type not be inferred via `ViewTraits::hooks_policy`?

Reply:
Thats what it does for the default case, but `customize_view_hooks()` allows another library (say KokkosResilience) to customize how the view hooks are created from the traits. Basically only the type is considered. So this struct is just a utility for getting the type from customize_view_hooks. It could be a alias but I kept it as a struct in case we needed to specialize it for some reason.